src/Shouldly/ShouldlyTimeoutException.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace Shouldly
+{
+    public class ShouldlyTimeoutException : Exception
```

Comment:
After some testing, I think it, in fact, would be better to extend TimeoutException. Not only for backwards compatibility, but also to enhance Shouldly's messages.


The following code
```
[Test]
public void ShouldHandleTimeoutException()
{
    var task = new Func<Task>(() => {System.Threading.Thread.Sleep(20000); 
                                     return new Task(null); });
    task.ShouldThrow(typeof(Exception));
}
```

With my code claims that "Shouldly.ShouldlyTimeoutException was thrown". 
When extending TimeoutException it says that the operation exceeded the time limit, which is clearer.